### PR TITLE
feat: add redisSub source and redisPub sink

### LINKED ISSUE #2247: Redis stream source
Support built-in redis source

### PR DESCRIPTION
**Pull request by retoool**
Add two new builtin connectors.

redisSub Source:
Add a new data source called "redisSub." This source empowers users to subscribe to specific Redis channels and stream received messages into Ekuiper. Configuration parameters are included to specify the Redis server and the channel to subscribe to.

redisPub Sink:
Add a new data sink called "redisPub." The redisPub sink allows users to publish data from Ekuiper to a designated Redis channel. Configuration parameters are included to specify the Redis server and the target channel.

Bug Fix:
Resolve a bug encountered during testing of the redisPub sink with ctx.TransformOutput() using a mock context at mock.RunSinkCollect(). The bug is fixed by adding context.TransKey to the context.NewMockContext(), ensuring that testing ctx.TransformOutput() no longer results in errors.

Closes #2247 

New PR of #2260

### CODE REVIEW

**Comment by ngjaying** [comment]
LGTM. Could you please add more UT to cover the Configure in redisPub and redisSub

**Comment by retoool** [comment]
> LGTM. Could you please add more UT to cover the Configure in redisPub and redisSub

Thank you for your input! I've already implemented unit tests and will be sure to include more to cover the Configure function in redisPub and redisSub.

File: internal/io/redis/pubsub/redisPub_test.go [hunk 1]
```diff
@@ -0,0 +1,190 @@
...
+
+func TestSinkPingRedisError(t *testing.T) {
+	s := RedisPub()
+	prop := map[string]interface{}{
+		"address": "",
+		"db":      0,
+		"channel": DefaultChannel,
+	}
+	// expErrStr := fmt.Sprintf("Ping Redis failed with error: %v\", err")
+	expErrStr := fmt.Sprintf("Ping Redis failed with error: dial tcp 127.0.0.1:6379: connectex: No connection could be made because the target machine actively refused it.")
```
Comment (ngjaying) [inline internal/io/redis/pubsub/redisPub_test.go:177]: Looks like the error message is different in local and Github env

**Review by ngjaying — CHANGES_REQUESTED** [review]

File: internal/io/redis/pubsub/redisPub_test.go (hunk 1 above)
Comment (retoool) [reply internal/io/redis/pubsub/redisPub_test.go]: Thank you for bringing this to my attention. It appears that there was a difference in the error message between the local environment and GitHub. I have investigated this issue further, and have already made the necessary changes to resolve it.

**Review by ngjaying — APPROVED** [review]
Cool, thanks very much!

**Pull request merged**